di-ipv-core-stub/src/main/java/uk/gov/di/ipv/stub/core/utils/HandlerHelper.java

```diff
@@ -128,20 +130,61 @@ public HTTPResponse sendHttpRequest(HTTPRequest httpRequest) {
         try {
             return httpRequest.send();
         } catch (IOException | SerializeException exception) {
-            logger.error("Failed to send a http request", exception);
+            LOGGER.error("Failed to send a http request", exception);
             throw new RuntimeException("Failed to send a http request", exception);
         }
     }
 
     public AuthorizationRequest createAuthorizationRequest(
-            State state, CredentialIssuer credentialIssuer, String jwt) {
+            State state, CredentialIssuer credentialIssuer, SignedJWT jwt) {
         return new AuthorizationRequest.Builder(
                         new ResponseType(ResponseType.Value.CODE),
                         new ClientID(CoreStubConfig.CORE_STUB_CLIENT_ID))
                 .state(state)
                 .scope(new Scope("openid"))
                 .redirectionURI(CoreStubConfig.CORE_STUB_REDIRECT_URL)
-                .customParameter("request", jwt)
+                .customParameter("request", jwt.serialize())
+                .endpointURI(credentialIssuer.authorizeUrl())
+                .build();
+    }
+
+    public AuthorizationRequest createAuthorizationJAR(
+            State state, CredentialIssuer credentialIssuer, SharedClaims sharedClaims, RSAKey signingKey) throws JOSEException {
+        Instant now = Instant.now();
+
+        ClientID clientID = new ClientID(CoreStubConfig.CORE_STUB_CLIENT_ID);
+        JWSHeader header = new JWSHeader.Builder(JWSAlgorithm.RS256)
+                .keyID(signingKey.getKeyID())
+                .build();
+
+        JWTClaimsSet authClaimsSet = new AuthorizationRequest.Builder(ResponseType.CODE, clientID)
+                .redirectionURI(CoreStubConfig.CORE_STUB_REDIRECT_URL)
+                .scope(new Scope("openid"))
+                .state(state)
+                .build()
+                .toJWTClaimsSet();
+
+        JWTClaimsSet.Builder claimsSetBuilder =
+                new JWTClaimsSet.Builder(authClaimsSet)
+                        .audience(credentialIssuer.id())
+                        .issuer(CoreStubConfig.CORE_STUB_JWT_ISS_CRI_URI)
+                        .expirationTime(Date.from(now.plus(1L, ChronoUnit.HOURS)))
+                        .notBeforeTime(Date.from(now))
```

Comment:
Do we need the `issueTime` object also to go along with the `expirationTime` and `notBeforeTime` ?

Shows up as `iat` in the jwt payload